gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversalTest.java

```diff
@@ -71,4 +80,89 @@ public void graphTraversalShouldHaveMethodsOfAnonymousGraphTraversal() {
             }
         }
     }
+
+    @Test
+    public void graphTraversalMethodsShouldGenerateCorrespondingBytecode() throws Exception {
+        final Random random = new Random();
```

Comment:
I use it from time to time during normal development and it is certainly used before a release. We can't have random in tests where we don't have a clue how to reproduce the error. It leads to tons of wasted time and frustration when it happens. Additionally, we already mostly ignore our automated build systems because of false failure and the aim should be to get rid of things that cause us to doubt failure in there - randomness is one of those things. 

That said, I admit that there are tests that benefit from randomness and this is a good candidate for one. I'd offer two options:

1. In the short term (for this PR), we log the seed that is being used to the build output. In that way, if there is a failure on this test we can edit the code, plug in the seed for the failure and presumably recreate the failure. 
2. In the long term (a new ticket), we look at doing this more generally - maybe `assertNonDeterministic` takes a seed instead of being binary true/false or we use [junit-quickcheck](http://pholser.github.io/junit-quickcheck/site/0.6.1/) which is built precisely for this kind of situation.

wdyt?